layout/save.go

```diff
@@ -17,69 +12,41 @@ func (i *Image) Save(additionalNames ...string) error {
 
 // SaveAs ignores the image `Name()` method and saves the image according to name & additional names provided to this method
 func (i *Image) SaveAs(name string, additionalNames ...string) error {
-	err := i.mutateCreatedAt(i.Image, v1.Time{Time: i.createdAt})
-	if err != nil {
-		return errors.Wrap(err, "set creation time")
-	}
-
-	if i.Image, err = imgutil.OverrideHistoryIfNeeded(i.Image); err != nil {
-		return fmt.Errorf("override history: %w", err)
+	if err := i.SetCreatedAtAndHistory(); err != nil {
```

Comment:
We have to do this at the point of save (vs instantiation) because otherwise we might modify the digest when there is a base image and that could be surprising